pkg/kv/kvserver/raft_transport.go

```diff
@@ -502,13 +517,14 @@ func (t *RaftTransport) SendAsync(
 	}
 
 	select {
-	case ch <- req:
+	case q.reqs <- req:
+		q.bytes.add(int64(req.Size()))
```

Comment:
I think there is a subtle bug possibility here.

It looks like requests are “released” into the pool (I guess, to avoid allocations). So, after we send `req`, it can be written to concurrently, hence it's not safe/correct to call `req.Size()` like this.

I'll put the `req.Size()` computation before the send, with a comment why.

Reply:
Makes sense. This would also make sense if we were to start using these for memory budgeting, where we would have to allocate budget before the send, and free it after the receive.

Reply:
I actually did hit this bug while testing locally. I observed the metric going negative. Probably what happened is that sometimes the "release" overtook and emptied the request, so `Add(req.Size())` was leaky (and reported a nearly-0 instead of real size), unlike the accurate `Add(-req.Size())` on the receiving end.